applications/RANSApplication/tests/cpp/test_vms_monolithic_k_based_condition.cpp

```diff
@@ -0,0 +1,345 @@
+    auto& r_element = r_model_part.Elements().front();
+    r_condition.SetValue(NEIGHBOUR_ELEMENTS, GlobalPointersVector<Element>{&r_element});
+
+    // set process info variables
+    auto& r_process_info = r_model_part.GetProcessInfo();
+    r_process_info.SetValue(WALL_VON_KARMAN, 3.1);
+    r_process_info.SetValue(WALL_SMOOTHNESS_BETA, 4.2);
+    r_process_info.SetValue(RANS_LINEAR_LOG_LAW_Y_PLUS_LIMIT, 12.0);
+    r_process_info.SetValue(TURBULENCE_RANS_C_MU, 0.09);
+
+    return r_model_part;
+}
+
+} // namespace
+
+KRATOS_TEST_CASE_IN_SUITE(RansVMSMonolithicKBasedWall2D2N_EquationIdVector, KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansVMSMonolithicKBasedWall2D2N_SetUp(model);
+
+    // Test:
+    auto eqn_ids = std::vector<IndexType>{};
+    for (const auto& r_condition : r_model_part.Conditions())
+    {
+        r_condition.EquationIdVector(eqn_ids, r_model_part.GetProcessInfo());
+        KRATOS_CHECK_EQUAL(eqn_ids.size(), r_condition.GetGeometry().PointsNumber() * 3);
+        const auto& r_geometry = r_condition.GetGeometry();
+        IndexType local_index = 0;
+        for (IndexType i_node = 0; i_node < r_geometry.PointsNumber(); ++i_node)
+        {
+            KRATOS_CHECK_EQUAL(eqn_ids[local_index++], r_geometry[i_node].Id() * 4);
+            KRATOS_CHECK_EQUAL(eqn_ids[local_index++], r_geometry[i_node].Id() * 4 + 1);
+            KRATOS_CHECK_EQUAL(eqn_ids[local_index++], r_geometry[i_node].Id() * 4 + 3);
+        }
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansVMSMonolithicKBasedWall2D2N_GetDofList, KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansVMSMonolithicKBasedWall2D2N_SetUp(model);
+
+    // Test:
+    auto dofs = Element::DofsVectorType{};
+    for (const auto& r_condition : r_model_part.Conditions())
+    {
+        r_condition.GetDofList(dofs, r_model_part.GetProcessInfo());
+        KRATOS_CHECK_EQUAL(dofs.size(), r_condition.GetGeometry().PointsNumber() * 3);
+        const auto& r_geometry = r_condition.GetGeometry();
+        IndexType local_index = 0;
+        for (IndexType i_node = 0; i_node < r_geometry.PointsNumber(); ++i_node)
+        {
+            KRATOS_CHECK_EQUAL(dofs[local_index]->GetVariable(), VELOCITY_X);
+            KRATOS_CHECK_EQUAL(dofs[local_index++]->EquationId(),
+                               r_geometry[i_node].Id() * 4);
+            KRATOS_CHECK_EQUAL(dofs[local_index]->GetVariable(), VELOCITY_Y);
+            KRATOS_CHECK_EQUAL(dofs[local_index++]->EquationId(),
+                               r_geometry[i_node].Id() * 4 + 1);
+            KRATOS_CHECK_EQUAL(dofs[local_index]->GetVariable(), PRESSURE);
+            KRATOS_CHECK_EQUAL(dofs[local_index++]->EquationId(),
+                               r_geometry[i_node].Id() * 4 + 3);
+        }
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansVMSMonolithicKBasedWall2D2N_CalculateLocalSystem, KratosRansFastSuite)
```

Comment:
Convention

Reply:
This I would like to keep as it is because its a long name, having underscores gives little bit of readability :)